### PR TITLE
21 login system

### PR DESCRIPTION
**Pull request by josephwehby**
# What I Did
1. created a Login class
- within this class there are the following methods: createUser, userInput, readCSV, outputCSV, and verifyUser
- additionally, it stores a file name and a map of the users keyed on the username with a val being a User object
2. created a User class
- within this class is the user constructor which takes in a username, password, and account type and based on the account type it assigns a permission level
- this class also has two friend classes: InventoryManager and Login 
3. wrote method in inventory manager that calls the Login userInput method and sets the current user
4.  added two additional variables to InventoryManager class: current_user pointer and Login class object pointer
5. updated makefile so everything compiles 

you should probably pull this to your own system and test it before merging

# Potential Change
- as of now the outputCSV method is called within the destructor. I know Noah mentioned he was not a fan of this in one of our scrums so I am open to moving the call elsewhere 
- currently the program only allows for three account types: owner, manager, employee but am open to adding more

### CODE REVIEW

File: src/Login.cpp [hunk 1]
```diff
@@ -0,0 +1,141 @@
+#include "Login.hpp"
+
+Login::Login() { /* does nothing */
+}
+
+Login::~Login() {
+    /* no deletion required*/
+    outputCSV();
```
Comment (Brenaren) [inline src/Login.cpp:8]: We had talked about not calling printouts in the destructor in favor of explicit prints.

**Review by Brenaren — CHANGES_REQUESTED** [review]
I left a comment about calling the print in the destructor, other than that it looks good!

File: src/Login.cpp (hunk 1 above)
Comment (josephwehby) [reply src/Login.cpp]: moved the statement into userLogin in IM

**Review by Brenaren — APPROVED** [review]
Looks good to me!

**Review by vbroda17 — APPROVED** [review]
Everything looks good. Did not notice anything that requires changing.

**Pull request merged**